dcurve/src/internal/clr.ts

```diff
@@ -0,0 +1,151 @@
+'use strict';
+
+import {
+  CLRArgs,
+  GrantMatch,
+  GrantPrediction,
+  GrantPredictionArgs,
+  GrantPredictions,
+  GrantRoundContributions,
+  GrantsDistribution,
+  InitArgs,
+  PayoutMatch,
+  PayoutMatches,
+} from '../../src/types';
+import { generateMerkle, getMerkleRoot } from './merkle';
+import { addAnonContribution, getGrantMatch } from './utils';
+
+// polyfill Buffer for use in browser
+import { Buffer } from 'buffer/';
+// set to window (TODO: make this work in vite.config.ts)
+Object.defineProperty(window, 'Buffer', { value: Buffer });
+
+export class CLR {
+  _options: InitArgs;
+
+  constructor(options: InitArgs) {
+    this._options = options;
+  }
+
+  calculate(grantRoundContributions: GrantRoundContributions, _options?: Record<string, unknown>): GrantsDistribution {
+    // allow the options to be overridden
+    const options = Object.assign(this._options, _options || {});
```

Comment:
This was done this way to allow the calc implementations to have an extended option set which could be defined both globally and on a per calc basis. 

This was maybe more important when we intended to allow the hash mechanism to be defined as an option, but I still think there may be extra options we'd like to pass in to some calculations and not every calculation (for example the pairwise algorithm accepts a `threshold` which is specific to the round, this way we don't need a new instance to calculate each distribution, we can just feed the `threshold` on `calculate`/`predict`).

You are right though, creating a new instance would suffice! I'm happy to go either way here - what would your preference be? @mds1 @thelostone-mc

Reply:
Got, thanks for the info. I'm fine with whatever you both think is best, since you're more familiar with the match algorithm/potential usages than I am!